src/renderer/splash.tsx

```diff
@@ -1,104 +1,81 @@
 import { Center, ChakraProvider, Flex, Progress, Text, VStack } from '@chakra-ui/react';
-import { memo, useEffect, useState } from 'react';
+import { memo, useCallback, useEffect, useState } from 'react';
 import { createRoot } from 'react-dom/client';
 import './i18n';
 import { useTranslation } from 'react-i18next';
+import { SetupStage } from '../common/backend-setup';
 import { ipcRenderer } from '../common/safeIpc';
+import { assertNever } from '../common/util';
 import { ChaiNNerLogo } from './components/chaiNNerLogo';
 import { theme } from './splashTheme';
 
 const Splash = memo(() => {
     const { t } = useTranslation();
 
-    const [status, setStatus] = useState(t('splash.loading', 'Loading...'));
+    const getStatusText = useCallback(
+        (stage: SetupStage): string => {
+            switch (stage) {
```

Comment:
Instead of this gigantic switch, couldn't we just send the text data with the event?

Reply:
I see why you did it this way (so you could use the translation hook), but I think you can just use i18next for that in the main process

Reply:
I guess that makes sense, I'm just thinking about it as the frontend shouldn't (imo) care about how the backend is implemented. With this (and the current code) if i had a backend that did different steps, the frontend wouldn't be able to display that

Reply:
> I'm just thinking about it as the frontend shouldn't (imo) care about how the backend is implemented.

Yeah, agreed. I'll move the switch into `main/`.

> if i had a backend that did different steps, the frontend wouldn't be able to display that

The backend also wouldn't compile because both the frontend and backend communicate to each other via a strictly type-checked API. If you want to add a new stage, TS will ensure that you add it everywhere.